test/integration.test.js

```diff
@@ -0,0 +1,68 @@
+const slowDown = require("../lib/express-slow-down");
+const express = require("express");
+const http = require("http");
+const axios = require("axios");
+const assert = require("assert");
+
+const TEST_PORT = 3300;
+const SERVER_WORKING_DELAY = 100;
+const defaultConfig = {
+  headers: true,
+};
+
+function assertHeaders(payload, limit, remaining) {
+  assert(Number(payload.headers["x-slowdown-remaining"]) === remaining);
+  assert(Number(payload.headers["x-slowdown-limit"]) === limit);
+}
+
+async function fetch() {
+  const { data, status, headers } = await axios(
+    `http://localhost:${TEST_PORT}`
+  );
+  return { data, status, headers };
+}
+
+function asyncServerClose(server) {
+  return new Promise((resolve, reject) => {
+    server.close((err) => {
+      if (err) {
+        reject(err);
+      } else resolve(true);
+    });
+  });
+}
+
+function buildTestServer(slowDownConfig) {
+  const app = express();
+  Object.assign(slowDownConfig, defaultConfig);
+  app.use(slowDown(slowDownConfig));
+  app.get("/", (req, res) => {
+    setTimeout(() => {
+      res.status(200);
+      res.send({ sucess: true });
+    }, SERVER_WORKING_DELAY);
+  });
+  return http.createServer(app).listen(TEST_PORT);
```

Comment:
Instead of hard-coding the port, it may be better to let the system assign on.

Example: https://gist.github.com/nfriedly/c3cf7a992a9543cd2e71df72843ae014#file-server-js-L28-L31